### PR TITLE
Changes login page

### PR DESCRIPTION
**Pull request by wariofdz**
### Changed:

1. Changed hook useLoginForm.
2. Added tags aria-label for elements in Login Page.
3. Created and added new thunk createAuthByCredentials in a new file, outside from authSlice file.
4. Added new barrel file from thunks.
5. Updated gitignore for ignore all of folder turbo repo.

### CODE REVIEW

**Comment by wariofdz** [comment]
### Last changed:

1. Changed useNavigate hook to Login component.
2. Updated version babel-jest and jest-environment-jsdom.
3. Added test Login component.
4. Added test useLoginForm hook.
5. Added test createAuthByCredentials thunk.
6. Changed test authSlice to include createAuthByCredentials thunk.

**Comment by wariofdz** [comment]
@notaphplover, could you revise the PR, please?

File: packages/frontend/web-ui/src/app/store/features/authSlice.js [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+function createAuthPendingReducer(state) {
+  state.status = 'checking';
+  state.token = null;
+  state.errorMessage = null;
+}
+
+function createAuthFulfilledReducer(state, action) {
+  state.status = 'not-authenticated';
+
+  if (action.payload.statusCode === 200) {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/features/authSlice.js:14]: Maybe a switch block would be more appropiate

File: packages/frontend/web-ui/src/app/store/features/authSlice.js [hunk 2]
```diff
@@ -1,6 +1,35 @@
 import { createSlice } from '@reduxjs/toolkit';
+import { createAuthByCredentials } from '../thunk/createAuthByCredentials';
 import { createAuthByToken } from '../thunk/createAuthByToken';
 
+function createAuthPendingReducer(state) {
+  state.status = 'checking';
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/features/authSlice.js:6]: should we declare a constant for this string value?

File: packages/frontend/web-ui/src/app/store/features/authSlice.js [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+  state.token = null;
+  state.errorMessage = null;
+}
+
+function createAuthFulfilledReducer(state, action) {
+  state.status = 'not-authenticated';
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/features/authSlice.js:12]: should we declare a constant for this string value?

File: packages/frontend/web-ui/src/app/store/thunk/createAuthByCredentials.spec.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import { afterAll, beforeAll, describe, expect, it } from '@jest/globals';
+import { createAuthByCredentials } from './createAuthByCredentials';
+import { httpClient } from '../../../common/http/services/HttpService';
+import { buildSerializableResponse } from '../../../common/http/helpers/buildSerializableResponse';
+
+jest.mock('../../../common/http/services/HttpService');
+jest.mock('../../../common/http/helpers/buildSerializableResponse');
+
+describe('thunk createAuthByCredentials', () => {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/thunk/createAuthByCredentials.spec.js:9]:
```suggestion
describe('createAuthByCredentials', () => {
```

File: packages/frontend/web-ui/src/app/store/thunk/index.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+export * from './createAuthByCredentials';
+export * from './createAuthByToken';
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/thunk/index.js:2]: why should we have this barrel module?

**Review by notaphplover — CHANGES_REQUESTED** [review]
Looks great, I just added some minor suggestions

File: packages/frontend/web-ui/src/app/store/thunk/index.js (hunk 5 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/thunk/index.js:2]: Ups :S I forgot to delete this file

File: packages/frontend/web-ui/src/app/store/thunk/createAuthByCredentials.spec.js (hunk 4 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/thunk/createAuthByCredentials.spec.js:9]: Done!

File: packages/frontend/web-ui/src/app/store/features/authSlice.js (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/features/authSlice.js:6]: Done! The name of this new const variable is **STATUS_AUTH_CHECKING**.

File: packages/frontend/web-ui/src/app/store/features/authSlice.js (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/features/authSlice.js]: Done!

File: packages/frontend/web-ui/src/app/store/thunk/index.js (hunk 5 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/thunk/index.js]: Done!

File: packages/frontend/web-ui/src/app/store/features/authSlice.js [hunk 6]
```diff
@@ -2,27 +2,34 @@ import { createSlice } from '@reduxjs/toolkit';
...
-  } else if (action.payload.statusCode === 422) {
-    state.errorMessage = 'Unprocessable operation. Try again.';
-  } else {
-    state.errorMessage = 'Ups... something strange happened. Try again?';
+  switch (action.payload.statusCode) {
+    case 200:
+      state.status = STATUS_AUTH_AUTHENTICATED;
+      state.token = action.payload.body.jwt;
+      break;
+    case 422:
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/app/store/features/authSlice.js:23]: Docs are wrong, there's no 422 response. Instead, there's a 401 response when credentials are wrong.

**Review by notaphplover — CHANGES_REQUESTED** [review]
Almost ready! :)

File: packages/frontend/web-ui/src/app/store/features/authSlice.js (hunk 6 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/features/authSlice.js:23]: Do I make the change now or I can wait to make later?

Comment (wariofdz) [reply packages/frontend/web-ui/src/app/store/features/authSlice.js:23]: Done!

**Review by notaphplover — APPROVED** [review]

**Pull request merged**